source/common/upstream/wrsq_scheduler.h

```diff
@@ -0,0 +1,170 @@
+#pragma once
+
+#include <algorithm>
+#include <iostream>
+#include <memory>
+#include <queue>
+#include <utility>
+#include <vector>
+
+#include "envoy/common/random_generator.h"
+#include "envoy/upstream/scheduler.h"
+
+#include "common/common/assert.h"
+
+#include "absl/container/flat_hash_map.h"
+
+namespace Envoy {
+namespace Upstream {
+
+// Weighted Random Selection Queue (WRSQ) Scheduler
```

Comment:
Looking at this comment after not thinking about this for 2 months, it can improve. Some observations to reference later:
* Elaborate on why it's like vanilla RR or WRS based on the weight distribution.
* Beef up the note.
* Consider an ASCII diagram.